model/api-models.js

```diff
@@ -3,6 +3,19 @@ const db = require("../db/connection");
 exports.selectTopics = () => {
     return db.query('SELECT * FROM topics')
         .then(({rows}) => {
-            return rows
-        })
+            return rows;
+        });
+};
+
+exports.selectArticleById = (article_id) => {
+    return db.query('SELECT * FROM articles WHERE article_id=$1', [article_id])
+    .then(({rows}) => {
+        if(rows.length === 0){
+           return Promise.reject({
+            status: 400,
+            msg: "Bad Request"
+           });
```

Comment:
We have handled this in the controller with the not-a-number check, therefore this check against rows.length isn't necessary. Either handle in the controller - or - in the model, but not both